src/components/SlotSelectorModal/SlotSelectorModal.tsx

```diff
@@ -0,0 +1,126 @@
+
+export interface SlotSelectorModalProps {
+  onDismiss: () => void;
+  isOpen: boolean;
+  packageName?: string;
+}
+const SlotSelectorModal: React.FC<SlotSelectorModalProps> = ({
+  onDismiss,
+  isOpen,
+  packageName,
+}) => {
+  const [selectedSlots, setSelectedSlots] = React.useState<string[]>([]);
+  const schema = useSubmissionSchema();
+  const { getHiddenSlotsForSchemaClass, setHiddenSlotsForSchemaClass } =
+    useStore();
+
+  const schemaClassName = packageName && TEMPLATES[packageName].schemaClass;
+  const templateName = packageName && TEMPLATES[packageName].displayName;
+
+  const slotGroups = useMemo(
+    () =>
+      schema.data && schemaClassName !== undefined
+        ? groupClassSlots(schema.data.schema, schemaClassName)
+        : [],
+    [schema.data, schemaClassName],
+  );
+  const allSlotNames = useMemo(
+    () => slotGroups.flatMap((group) => group.slots.map((s) => s.name)),
+    [slotGroups],
+  );
+
+  // This translates a list of hidden slots from the store into a list of selected slots for the
+  // SlotSelector component. If there are no hidden slots, all slots are selected by default. The
+  // isOpen state is used to reset the selected slots when the modal is closed (i.e. don't keep
+  // changes if the user cancels out of the modal).
+  useEffect(() => {
+    if (isOpen) {
+      const hiddenSlotsFromStore =
+        getHiddenSlotsForSchemaClass(schemaClassName);
+      if (hiddenSlotsFromStore === undefined) {
+        setSelectedSlots(allSlotNames);
+      } else {
+        setSelectedSlots(
+          allSlotNames.filter((s) => !hiddenSlotsFromStore.includes(s)),
+        );
+      }
+    } else {
+      setSelectedSlots([]);
+    }
+  }, [getHiddenSlotsForSchemaClass, isOpen, schemaClassName, allSlotNames]);
+
+  // When the user taps the Save button, translate the selected slots back into a list of hidden
+  // slots and save them to the store. Then close the modal.
+  const handleSave = () => {
+    if (schemaClassName !== undefined) {
+      const hiddenSlots = allSlotNames.filter(
+        (s) => !selectedSlots.includes(s),
+      );
+      setHiddenSlotsForSchemaClass(schemaClassName, hiddenSlots);
+    }
+    onDismiss();
+  };
+
+  return (
+    <IonModal
+      isOpen={isOpen}
+      onIonModalWillDismiss={onDismiss}
+      className={styles.slotSelectorModal}
+    >
+      <IonHeader>
+        <IonToolbar>
+          <IonButtons slot="start">
+            <IonButton color="medium" onClick={onDismiss}>
+              Cancel
+            </IonButton>
+          </IonButtons>
+          <IonTitle>Select Fields</IonTitle>
+          <IonButtons slot="end">
+            <IonButton onClick={handleSave} strong={true}>
+              Save
+            </IonButton>
+          </IonButtons>
+        </IonToolbar>
+      </IonHeader>
+      <IonContent>
+        <div className="ion-padding">
+          <p>
+            Select the fields you would like to see by default when entering
```

Comment:
I'm a little confused about the word "default" here. Maybe it's there to tell the user that it will be the new normal, but they can always change it later. Assuming I understand the user-facing behavior, I think I'd find this more clear:

```diff
- Select the fields you would like to see by default when entering
+ Select the fields you would like to see when entering
```

Reply:
I think I was trying to communicate that the preference is "sticky" across all existing and new samples for that template -- as opposed to a sample-by-sample thing. But, yeah, looking back at this it's a little confusing. I'll simplify it.